fix: set PostHog ui_host for toolbar auth

seanoliver opened the pull request:
**Problem**
Clicking “Authenticate” in the PostHog toolbar on supabase.com was sending us to `ph.supabase.com/.../authorize_and_redirect` and landing on a 404. That blocks us from using the toolbar for debugging/validation (which is basically the whole point of turning it on).

**What happened**
Our `ph.supabase.com` setup is an ingestion proxy (it forwards to `*.i.posthog.com`). The toolbar auth flow isn’t served off the ingestion host though — it lives on the PostHog UI host (`eu.posthog.com` / `app.posthog.com`). So the proxy was doing the “right” thing for events, but the toolbar login URL just doesn’t exist upstream.

**Changes**
- Set `ui_host` in our PostHog JS init config so toolbar auth routes to the UI host
- Keep `api_host` unchanged so events/decide traffic still goes through our proxy

**Testing**
- Ran prettier on `packages/common/posthog-client.ts`
- Ran `tsc --noEmit` for `packages/common`
- Confirmed `ph.supabase.com/authorize_and_redirect` 404s while `eu.posthog.com/authorize_and_redirect` redirects to login (expected)

Resolves GROWTH-577


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Added configurable UI host support for PostHog integration, allowing customization of the analytics platform endpoint.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

seanoliver committed:
fix: set PostHog ui_host for toolbar auth

seanoliver committed:
Merge remote-tracking branch 'origin/master' into fix/posthog-toolbar-ui-host

awaseem commented on packages/common/posthog-client.ts:
```diff
@@ -22,10 +23,13 @@ class PostHogClient {
-      apiHost:
-        config.apiHost || process.env.NEXT_PUBLIC_POSTHOG_HOST || 'https://ph.supabase.green',
+      apiHost,
+      uiHost: config.uiHost || process.env.NEXT_PUBLIC_POSTHOG_UI_HOST || 'https://eu.posthog.com',
```
minor nit, can we create a variable for ui host as well?

awaseem reviewed (approved):
Minor nit

seanoliver committed:
Extract uiHost as const for consistency

Matches the pattern used for apiHost, making the constructor
cleaner and more maintainable.

seanoliver committed:
Merge branch 'master' into fix/posthog-toolbar-ui-host
